src/legacy/core_plugins/data/public/index_patterns/fields/field_list.ts

```diff
@@ -17,24 +17,69 @@
  * under the License.
  */
 
-import { IndexedArray } from 'ui/indexed_array';
 import { NotificationsSetup } from 'kibana/public';
+import { findIndex } from 'lodash';
 import { IndexPattern } from '../index_patterns';
-import { Field, FieldSpec } from './field';
+import { Field, FieldType, FieldSpec } from './field';
 
-export class FieldList extends IndexedArray<Field> {
+type FieldMap = Map<Field['name'], Field>;
+
+export interface FieldListInterface extends Array<Field> {
+  getByName(name: Field['name']): Field | undefined;
+  getByType(type: Field['type']): Field[];
+  add(field: FieldSpec): void;
+  remove(field: FieldType): void;
+}
+
+export class FieldList extends Array<Field> implements FieldListInterface {
```

Comment:
nit: can we loose extends Array<Field> as thats already part of FieldListInterface definition ?

Reply:
It appears to be needed. Seems that FieldList doesn't inherit that trait from FieldListInterface.